lib/roast/workflow/base_workflow.rb

```diff
@@ -87,20 +87,40 @@ def chat_completion(**kwargs)
         })
 
         result
+      rescue Faraday::ResourceNotFound => e
+        execution_time = Time.now - start_time
+        message = e.response.dig(:body, "error", "message") || e.message
+        error = Roast::ResourceNotFoundError.new(message)
+        e.set_backtrace([])
+        log_and_raise_error(error, message, model, kwargs, execution_time)
+      rescue NoMethodError => e
+        execution_time = Time.now - start_time
+
+        if e.message.include?("undefined method 'chat' for nil")
```

Comment:
Can we catch this one earlier? IMO it should never get here.

Reply:
I've moved the logic to much earlier in the code for catching this sort of misconfiguration.